app/controllers/posts_controller.rb

```diff
@@ -14,10 +22,10 @@ def new
   end
 
   def create
-    @post = current_user.posts.build(params.require(:post).permit(:Title, :Text))
+    @post = current_user.posts.build(post_params)
     if @post.save
       flash[:success] = 'Post saved successfully'
-      redirect_to user_posts_path
+      redirect_to user_posts_path(@post.author_id)
     else
       flash.now[:error] = 'Post not saved, check your Entries'
       render :new
```

Comment:
- [ ] I took the pleasure of trying to create a Post even after skipping the **_CRSF token_** check and I got this error most times:

![image](https://github.com/ismayito/Rails-blog-app/assets/77106962/b04c78db-1c88-4d72-9efa-5176bcc3bc67)

--- 
Remember that you have to explicitly state the `Author`, `PostsCounter` and `CommentsCounter` for a `Post` instance before it can be created. To do this, I will advise you to make your create action in the `Posts` controller like this:

```ruby
  def create
    @user = User.find(params[:user_id])
    @post = Post.new(post_params)
    @post.author_id = @user.id
    @post.CommentsCounter = 0
    @post.LikesCounter = 0
    if @post.save
      flash[:success] = 'Post saved successfully'
      redirect_to user_posts_path(@post.author_id)
    else
      flash.now[:error] = 'Post not saved, check your Entries'
      render :new
    end
  end
```